model_interpreter/interpreter.py

```diff
@@ -244,22 +247,32 @@ def _get_single_model_contribution(
 
         shap_vals = self.explainer.shap_values(single_row)
 
-        if isinstance(shap_vals, list):
-            if predict_class_index not in np.arange(
-                len(shap_vals) + 1
-            ) or not isinstance(predict_class_index, int):
-                raise ValueError("predicted class not a valid int")
-
-            shap_values = shap_vals[predict_class_index][0]
-
+        # if values are all 0, shap doesn't bother splitting by class which impacts
```

Comment:
Not really a fan of the logical flow in this code - very hard to follow. I suspect that it should be handled via generic dispatch on the basis of model type. Will create a follow up ticket

Reply:
I'm essentially having to rely on the tests. The code seems fine, but the use of magic dimension indicies feels prime for logic errors. But without in depth knowledge of the shap API I find it hard to debug. 

With passing tests and suitable test logic, I'm inclined to accept changes (pending the two small fix items in the review). Clearer code as a follow up ticket.